src/puppetlabs/trapperkeeper/services/status/status_core.clj

```diff
@@ -149,9 +161,17 @@
 
 (schema/defn ^:always-validate get-jvm-metrics :- JvmMetricsV1
   []
-  (let [runtime-bean (ManagementFactory/getRuntimeMXBean)]
+  (let [runtime-bean (ManagementFactory/getRuntimeMXBean)
+        gc-beans (jmx/mbean-names "java.lang:name=*,type=GarbageCollector")]
     {:heap-memory (jmx/read "java.lang:type=Memory" :HeapMemoryUsage)
      :non-heap-memory (jmx/read "java.lang:type=Memory" :NonHeapMemoryUsage)
+     :file-descriptors (clojure.set/rename-keys
+                        (jmx/read "java.lang:type=OperatingSystem" [:OpenFileDescriptorCount :MaxFileDescriptorCount])
+                        {:OpenFileDescriptorCount :used :MaxFileDescriptorCount :max})
+     :gc-stats (map #(conj {:name (.getCanonicalName %)}
+                           (clojure.set/rename-keys (jmx/read % [:CollectionCount :CollectionTime])
+                                                    {:CollectionCount :count :CollectionTime :total-time-ms}))
+                    gc-beans)
```

Comment:
Yup. It currently returns a list of `[{:name, :count, :total-time-ms}]`. Would be great to consolidate that into a map like `{:young-gc {...}, :tenured-gc {...}}`. But:

  - We'd need to find or maintain a mapping of GC algorithm names -> tenured/young.

  - This assumes there's always two GC algorithms. This appears to be the case for the default Scavange/MarkSweep and G1GC, but I'm not sure if it holds for all GC options.

Looks like Dropwizard Metrics returns a map of GC algorithm name -> metric set:

https://github.com/dropwizard/metrics/blob/v3.1.2/metrics-jvm/src/main/java/com/codahale/metrics/jvm/GarbageCollectorMetricSet.java

Could look into doing something similar.